ci_visibility.go

```diff
@@ -0,0 +1,257 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type EnvConfig struct {
+	EnvNames     []string
+	DefaultValue string
+	IsRequired   bool
+}
+
+var pipelineVisibilityEnvConfigs = map[string]EnvConfig{
+	"pipeline_end": {
+		EnvNames:     []string{"PLUGIN_BUILD_FINISHED", "DRONE_BUILD_FINISHED", "CI_BUILD_FINISHED"},
```

Comment:
just realised the current implementation uses `DRONE_BUILD_FINISHED` for the pipeline end time, which according to Harness docs is set to the same value as `DRONE_BUILD_CREATED` for running builds. This means we're sending the same timestamp for both start and end times to Datadog, making the pipeline appear to have zero duration.

Since this plugin runs in a specific stage, we should use `DRONE_STAGE_FINISHED` instead. This will give us the actual end time of the stage where the plugin is executing, which is more accurate for tracking the plugin's execution time or the user could make use of the input `PLUGIN_BUILD_FINISHED`